pkg/provisioner/ironic/factory.go

```diff
@@ -196,3 +261,153 @@ func loadTLSConfigFromEnv() clients.TLSConfig {
 		SkipClientSANVerify:   skipClientSANVerify,
 	}
 }
+
+func (f *ironicProvisionerFactory) loadConfigFromIronicCR(ctx context.Context) (endpoint string, auth clients.AuthConfig, tlsConfig clients.TLSConfig, err error) {
+	// Get the Ironic CR
+	ironicCR := &ironicv1alpha1.Ironic{}
+	key := types.NamespacedName{
+		Name:      f.ironicName,
+		Namespace: f.ironicNamespace,
+	}
+
+	err = f.k8sClient.Get(ctx, key, ironicCR)
+	if err != nil {
+		return "", clients.AuthConfig{}, clients.TLSConfig{}, fmt.Errorf("failed to get Ironic CR %s/%s: %w", f.ironicNamespace, f.ironicName, err)
+	}
+
+	// Check if the Ironic CR is ready
+	ready := false
+	for _, condition := range ironicCR.Status.Conditions {
+		if condition.Type == "Ready" && condition.Status == metav1.ConditionTrue {
+			ready = true
+			break
+		}
+	}
+	if !ready {
+		return "", clients.AuthConfig{}, clients.TLSConfig{}, fmt.Errorf("ironic CR %s/%s is not ready", f.ironicNamespace, f.ironicName)
+	}
+
+	// Construct endpoint URL
+	apiPort := ironicCR.Spec.Networking.APIPort
+	if apiPort == 0 {
+		apiPort = 6385 // default port
```

Comment:
This is probably not correct: the service endpoint uses the default HTTP(s) port.